06x-new-kafka-roller.md

```diff
@@ -0,0 +1,331 @@
+ - <i>reason</i>: It is updated based on the current predicate logic from the Reconciler. For example, an update in the Kafka CR is detected.
+ - <i>numRestartAttempts</i>: The value is incremented each time the node has been attempted to restart.
+ - <i>numReconfigAttempts</i>: The value is incremented each time the node has been attempted to reconfigure.
+ - <i>numRetries</i>: The value is incremented each time the node cannot be restarted/reconfigured due to not meeting safety conditions (more on this later).
+ - <i>lastTransitionTime</i>: System.currentTimeMillis of last observed state transition.
+
+ <b>States</b>
+ | State        | Description |
+ | :-------------------- | :---------- |
+ | UNKNOWN               | The initial state when creating `Context` for a node. We expect to transition from this state fairly quickly after creating the context for nodes.  |
+ | NOT_RUNNING           | Node is not running (Kafka process is not running) |
+ | NOT_READY             | Node is running but not ready to serve requests (broker state < 2 OR == 127) |
+ | RESTARTED             | After successful `kubectl delete pod`. |
+ | RECONFIGURED          | After successful Kafka node config update via Admin client. |
+ | RECOVERING            | Node has started but is in log recovery (broker state == 2). |
+ | SERVING               | Node is in running state and ready to serve requests (broker state >= 3 AND != 127). |
+ | LEADING_ALL_PREFERRED | Node is in running state and leading all preferred replicas. |
+
+The broker states are defined [here](https://github.com/apache/kafka/blob/58ddd693e69599b177d09c2e384f31e7f5e11171/metadata/src/main/java/org/apache/kafka/metadata/BrokerState.java#L46).
+
+### Configurability
+The following can be the configuration options for the new KafkaRoller:
+
+| Configuration | Default value | Exposed to user | Description |
+| :-------------| :-------------| :---------------| :---------- |
+| maxRestartAttempts | 3 | No | The maximum number of times a node can be restarted before failing the reconciliation. This is checked against the node's `numRestartAttempts`.|
+| maxReconfigAttempts | 3 | No | The maximum number of times a node can be reconfigured before restarting it. This is checked against the node's `numReconfigAttempts`.|
+| maxRetries| 10 | No | The maximum number of times a node can retried after not meeting the safety conditions. This is checked against the node's `numRetries`.|
+| postOperationTimeoutMs | 60 seconds | Yes | The maximum amount of time we will wait for nodes to transition to `SERVING` state after an operation in each retry. This will be based on the operation timeout that is already exposed to the user via environment variable `STRIMZI_OPERATION_TIMEOUT_MS`. |
+| maxRestartParallelism | 1 | Yes | The maximum number of broker nodes that can be restarted in parallel. |
```

Comment:
```suggestion
| Configuration          | Default value | Exposed to user | Description                                                                                                                                                                                                                                                   |
|:-----------------------|:--------------|:----------------|:--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| maxRestartAttempts     | 3             | No              | The maximum number of times a node can be restarted before failing the reconciliation. This is checked against the node's `numRestartAttempts`.                                                                                                               |
| maxReconfigAttempts    | 3             | No              | The maximum number of times a node can be reconfigured before restarting it. This is checked against the node's `numReconfigAttempts`.                                                                                                                        |
| maxRetries             | 10            | No              | The maximum number of times a node can retried after not meeting the safety conditions. This is checked against the node's `numRetries`.                                                                                                                      |
| postOperationTimeoutMs | 60 seconds    | Yes             | The maximum amount of time we will wait for nodes to transition to `SERVING` state after an operation in each retry. This will be based on the operation timeout that is already exposed to the user via environment variable `STRIMZI_OPERATION_TIMEOUT_MS`. |
| maxRestartParallelism  | 1             | Yes             | The maximum number of broker nodes that can be restarted in parallel.                                                                                                                                                                                         |
```